opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1654,13 +1654,9 @@ namespace Opm
             segment_densities_[seg] = density / volrat;
 
             // calculate the mass rates
-            // TODO: for now, we are not considering the upwinding for this amount
-            // since how to address the fact that the derivatives is not trivial for now
-            // and segment_mass_rates_ goes a long way with the frictional pressure loss
-            // and accelerational pressure loss, which needs some work to handle
             segment_mass_rates_[seg] = 0.;
             for (int comp_idx = 0; comp_idx < num_components_; ++comp_idx) {
-                const EvalWell rate = getSegmentRate(seg, comp_idx);
+                const EvalWell rate = getSegmentRateUpwinding(seg, comp_idx);
                 segment_mass_rates_[seg] += rate * surf_dens[comp_idx];
```

Comment:
I think we should add a note to the segment_mass_rates_ data member that its elements have derivatives from mixed sources. Perhaps better, make an explanation of the approach in one place, and refer to it from all other places where it is necessary to have that understanding (such as where we modify duneD_ in the modified code below).

Reply:
There is already an explanation in getSegmentRateUpwinding. Do you want more?

Reply:
No, that is sufficient, thanks for pointing it out. I do think we should refer to it or mark the variables somehow (with comments, not naming) so that we can easily tell which variables are mixed.